vectordraw/public/js/studio_edit.js

```diff
@@ -0,0 +1,139 @@
+        $field.bind("change input paste", fieldChanged);
+        $resetButton.click(function() {
+            $field.val($wrapper.attr('data-default')); // Use attr instead of data to force treating the default value as a string
+            $wrapper.removeClass('is-set');
+            $resetButton.removeClass('active').addClass('inactive');
+        });
+        if (type == 'html' && tinyMceAvailable) {
+            tinyMCE.baseURL = baseUrl + "/js/vendor/tinymce/js/tinymce";
+            $field.tinymce({
+                theme: 'modern',
+                skin: 'studio-tmce4',
+                height: '200px',
+                formats: { code: { inline: 'code' } },
+                codemirror: { path: "" + baseUrl + "/js/vendor" },
+                convert_urls: false,
+                plugins: "link codemirror",
+                menubar: false,
+                statusbar: false,
+                toolbar_items_size: 'small',
+                toolbar: "formatselect | styleselect | bold italic underline forecolor wrapAsCode | bullist numlist outdent indent blockquote | link unlink | code",
+                resize: "both",
+                setup : function(ed) {
+                    ed.on('change', fieldChanged);
+                }
+            });
+        }
+
+    });
+
+    var studio_submit = function(data) {
+        var handlerUrl = runtime.handlerUrl(element, 'submit_studio_edits');
+        runtime.notify('save', {state: 'start', message: gettext("Saving")});
+        $.ajax({
+            type: "POST",
+            url: handlerUrl,
+            data: JSON.stringify(data),
+            dataType: "json",
+            global: false  // Disable Studio's error handling that conflicts with studio's notify('save') and notify('cancel') :-/
+        }).done(function(response) {
+            runtime.notify('save', {state: 'end'});
+        }).fail(function(jqXHR) {
+            if (jqXHR.responseText) { // Is there a more specific error message we can show?
+                try {
+                    errorMessage = JSON.parse(jqXHR.responseText).error;
+                    if (typeof errorMessage === "object" && errorMessage.messages) {
+                        // e.g. {"error": {"messages": [{"text": "Unknown user 'bob'!", "type": "error"}, ...]}} etc.
+                        errorMessage = $.map(errorMessage.messages, function(msg) { return msg.text; }).join(", ");
+                    }
+                } catch (error) { errorMessage = jqXHR.responseText.substr(0, 300); }
+            }
+            runtime.notify('error', {title: gettext("Unable to update settings"), message: errorMessage});
+        });
+    };
+
+    return {
+
+        getContents: function(fieldName) {
+            return _.findWhere(fields, {name: fieldName}).val();
+        },
+
+        save: function(data) {
+            var values = {};
+            var notSet = []; // List of field names that should be set to default values
+            for (var i in fields) {
```

Comment:
Why not use [`_.each()`](http://underscorejs.org/#each) or [`$.each()`](http://api.jquery.com/jQuery.each/) instead?